### PR TITLE
Enable proguard for release build

### PR DESCRIPTION
**Pull request by Ilya-Gh**
This PR enables proguard for release build as requested in https://github.com/artem-zinnatullin/qualitymatters/issues/156

### CODE REVIEW

File: proguard/proguard-square-okhttp.pro [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# OkHttp
+-keepattributes Signature
+-keepattributes *Annotation*
+-keep class com.squareup.okhttp.** { *; }
```
Comment (vanniktech) [inline proguard/proguard-square-okhttp.pro]: OkHttp does not require any rules.

File: proguard/proguard-square-retrofit2.pro [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Retrofit 2.X
+## https://square.github.io/retrofit/ ##
```
Comment (vanniktech) [inline proguard/proguard-square-retrofit2.pro]: Check again on the webiste, those rules are outdated

File: proguard/proguard-rxjava.pro [hunk 3]
```diff
@@ -0,0 +1,17 @@
+-dontwarn sun.misc.**
+
+-keepclassmembers class rx.internal.util.unsafe.*ArrayQueue*Field* {
```
Comment (artem-zinnatullin) [inline proguard/proguard-rxjava.pro]: I vote for lib :)

File: app/proguard-rules.pro [hunk 4]
```diff
@@ -0,0 +1,2 @@
+-keep class com.artemzin.qualitymatters.models.** { *; }
```
Comment (artem-zinnatullin) [inline app/proguard-rules.pro]: Not required

File: app/proguard-rules.pro [hunk 5]
```diff
@@ -0,0 +1,2 @@
...
+-keep class com.artemzin.qualitymatters.api.entities.** { *; }
```
Comment (artem-zinnatullin) [inline app/proguard-rules.pro]: Can we keep only annotated fields? Why keep class names/etc?

Comment (Ilya-Gh) [inline app/proguard-rules.pro]: AutoValue does not require any proguard rules so I think we can get rid of this.

**Comment by caipivara** [comment]
Guys check this https://github.com/yongjhih/android-proguards

**Comment by Ilya-Gh** [comment]
In my opinion it would be cool to have the possibility to update proguard rules by ourself without waiting for a 3rd party lib to be updated. @artem-zinnatullin What do you think about the lib mentioned about?

**Comment by artem-zinnatullin** [comment]
Well, I'm fine with small scoped libraries that provide proguard rules for specific libs.

**Comment by Ilya-Gh** [comment]
Added the libraries instead of proguard files, @artem-zinnatullin could you please check.

File: dependencies.gradle [hunk 6]
```diff
@@ -77,6 +80,12 @@ ext.libraries = [
         retrofitRxJavaAdapter   : "com.squareup.retrofit2:adapter-rxjava:$versions.retrofit",
         gson                    : "com.google.code.gson:gson:$versions.gson",
 
+        rxJavaProguardRules     : "com.artemzin.rxjava:proguard-rules:$versions.rxJavaProguardRules",
+        retrolambdaProguardRules: "com.github.yongjhih.android-proguards:retrolambda-pro:$versions.proguardRules",
+        retrofitProguardRules   : "com.github.yongjhih.android-proguards:retrofit2-pro:$versions.proguardRules",
```
Comment (vanniktech) [inline dependencies.gradle]: Use our own files /shrug

File: build.gradle [hunk 7]
```diff
@@ -50,6 +50,7 @@ buildscript {
 allprojects {
     repositories {
         jcenter()
+        maven { url 'https://jitpack.io' }
```
Comment (artem-zinnatullin) [inline build.gradle]: not a fan of this thing, but ok

File: dependencies.gradle (hunk 6 above)
Comment (artem-zinnatullin) [inline dependencies.gradle]: Mmmmmmm, wait, they all have one version? This is super weird and if so — I'm voting against these libs

**Comment by Ilya-Gh** [comment]
Reverted the branch to user proguard files with Rx rules defined in https://github.com/artem-zinnatullin/RxJavaProGuardRules.

**Comment by artem-zinnatullin** [comment]
@Ilya-Gh thank you for PR and sorry that it took so long to review!

**Pull request merged**